packages/@react-stately/selection/src/SelectionManager.ts

```diff
@@ -173,7 +173,7 @@ export class SelectionManager implements MultipleSelectionManager {
     let first: Node<unknown> | null = null;
     for (let key of this.state.selectedKeys) {
       let item = this.collection.getItem(key);
-      if (!first || item?.index < first.index) {
+      if (!first || (item && compareNodeOrder(this.collection, item, first) < 0)) {
```

Comment:
This accounts for the new collection implementation using indexes of items in sections relative to the parent node rather than the whole collection.